### DISCUSSION TITLE
PGEDC virtual meetup

### DISCUSSION THREAD

**Discussion by yrashk** [pgedc/pgedc#33] [Ideas]
Fellow extension developers,

I suggest organizing virtual meetups to provoke interesting conversations and socialize. 

For the inaugural meetup, perhaps we can just define a topic or two of interest and see how it goes. Perhaps we can do one or two short talks followed by discussions in the future.

What do you think about the idea of this meetup? Any suggestions for the topics?

--

Cc: @pgedc/members

**Comment by danolivo** [comment]
I guess one of the most tedious problems is registering hooks and collaborating with (at least identifying) other extensions, which hacks the core structures. We need it to prevent interference between libraries.
One more issue—having some objects registered in the database via CREATE EXTENSION, is it possible for a library to detect the moment of creation/deletion of these objects transactionally in some simple manner?
I would be glad to discuss this topic, but I am not ready to prepare a short talk.

**Reply by yrashk** [reply]
> I guess one of the most tedious problems is registering hooks and collaborating with (at least identifying) other extensions, which hacks the core structures. We need it to prevent interference between libraries. 

Having worked on it, I love this topic and would happily participate in a discussion.

> One more issue—having some objects registered in the database via CREATE EXTENSION, is it possible for a library to detect the moment of creation/deletion of these objects transactionally in some simple manner?

Can you elaborate on what exactly you mean by "detect"?

**Reply by danolivo** [reply]
Imagine the extension has a static library part and modifies core logic (in my case, it is usually planning). At the same time, it provides some UI with a set of functions and stores data in table format.
Tables are sometimes not a good solution for an extension, but they allow the extension to be transactional and work only in databases where 'CREATE EXTENSION' has been called. Also, the user can modify extension data in a regular way - by updating the tables.
In this situation, the extension on each call of its hooks should check the existence of the extension. 
As far as pg_extension doesn't have a syscache, searching through this table on each query adds a painfully large amount of overhead. Also, it would be better to register a callback on DROP EXTENSION rather than a hook on object modification.

**Comment by andreasscherbaum** [comment]
Strongly suggest to have a topic for the meetup, and then tack on a section for other questions.
One possible topic can be the binary distribution David suggested.
